provd/proto/identity.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+package identity;
+
+service IdentityService {
+  rpc GetIdentity(Empty) returns (GetIdentityResponse);
+  rpc SetIdentity(SetIdentityRequest) returns (Empty);
+  rpc ValidateUsername(ValidateUsernameRequest) returns (ValidateUsernameResponse);
```

Comment:
Whats the purpose of this call? Ie we validate username but not password / hostname / realname?